libs/community/langchain_community/vectorstores/falkordb_vector.py

```diff
@@ -0,0 +1,1684 @@
+        ssl: bool  = False,
+        pre_delete_collection: bool = False,
+    ) -> None:
+        try:
+            import falkordb 
+        except ImportError:
+            raise ImportError(
+                "Could not import falkordb python package."
+                "Please install it with `pip install falkordb`"
+            )
+        
+        # Only euclidean distance strategies is avaliable in falkor for now
+        if distance_strategy != DistanceStrategy.EUCLIDEAN_DISTANCE:
+            raise ValueError(
+                "distance_strategy must be 'EUCLIDEAN_DISTANCE'"
+            )
+        
+        # Graph object takes precedent over env or input params
+        if graph:
+            self._database = graph._graph
+            self._driver = graph._driver
+        else:
+            # Handle credentials via environment variables or input params
+            self._host = host
+            self._port = port
+            self._username = username or os.environ.get("FALKORDB_USERNAME")
+            self._password = password or os.environ.get("FALKORDB_PASSWORD")
+            self._ssl = ssl
+
+            # Initialize the FalkorDB connection
+            # Initialize the FalkorDB connection
+            try:
+                self._driver = falkordb.FalkorDB(
+                    host=self._host,
+                    port=self._port,
+                    username=self._username,
+                    password=self._password,
+                    ssl=self._ssl,
+                )
+            except redis.exceptions.ConnectionError:
+                raise ValueError(
+                    "Could not connect to FalkorDB database."
+                    "Please ensure that the host and port is correct"
+                )
+            except redis.exceptions.AuthenticationError:
+                raise ValueError(
+                    "Could not connect to FalkorDB database. "
+                    "Please ensure that the username and password are correct"
+                )
+            
+            # Verify that required values are not null
+            check_if_not_null(
+                ["embedding_node_property","node_label"],
+                [embedding_node_property, node_label],
+            )
+
+            print("Database name: ", database)
+            self._database = self._driver.select_graph(database)
+            self.database_name = database
+
+            self.embedding = embedding
+            self.node_label = node_label
+            self.relation_type = relation_type
+            self.embedding_node_property = embedding_node_property
+            self.text_node_property = text_node_property
+            self._distance_strategy = distance_strategy
+            self.override_relevance_score_fn = relevance_score_fn
+            #self.logger = logger
+            self.pre_delete_collection = pre_delete_collection
+            self.retrieval_query = retrieval_query
+            #self.relevance_score_fn = relevance_score_fn
```

Comment:
can be deleted?

Reply:
Deleted `#self.relevance_score_fn = relevance_score_fn`